Library/Homebrew/test/rubocops/cask/stanza_order_spec.rb

```diff
@@ -474,4 +474,65 @@
 
     include_examples "does not report any offenses"
   end
+
+  context "when `on_arch` blocks are out of order" do
+    let(:source) do
+      <<~CASK
+        cask 'foo' do
+          on_intel do
+            url "https://foo.brew.sh/foo-intel.zip"
+            sha256 :no_check
+            version :latest
+          end
+
+          on_arm do
+            url "https://foo.brew.sh/foo-arm.zip"
+            sha256 :no_check
+            version :latest
+          end
+
+          name "Foo"
+        end
+      CASK
+    end
+
+    let(:expected_offenses) do
+      [{
+        message:  "`on_intel` stanza out of order",
+        severity: :convention,
+        line:     2,
+        column:   2,
+        source:   "on_intel do\n    url \"https://foo.brew.sh/foo-intel.zip\"\n    sha256 :no_check\n    version :latest\n  end", # rubocop:disable Layout/LineLength
```

Comment:
Could probably make this and the one below heredocs or something to avoid the LineLength RuboCop violation, but I tried and the offense expectation failed related to indentation width no matter what kind of heredoc I used or what indentation I put in.

Reply:
Yeah, the span is between `on_intel` and `end`, so the indention of `on_intel` is not included.

Reply:
You would have to indent everything except the first line in the heredoc, which would also look kinda weird.

Reply:
```ruby
<<~EOS.chomp
on_intel do
    url "https://foo.brew.sh/foo-intel.zip"
    sha256 :no_check
    version :latest
  end
EOS
```